Spark failing in the CI

Spark tests are failing in master, see: https://github.com/ibis-project/ibis/runs/2356967362?check_suite_focus=true#step:4:137

This is the error:
```
E                   py4j.protocol.Py4JJavaError: An error occurred while calling None.org.apache.spark.api.java.JavaSparkContext.
E                   : java.net.BindException: Cannot assign requested address: Service 'sparkDriver' failed after 16 retries (on a random free port)! Consider explicitly setting the appropriate binding address for the service 'sparkDriver' (for example spark.driver.bindAddress for SparkDriver) to the correct binding address.
E                   	at sun.nio.ch.Net.bind0(Native Method)
E                   	at sun.nio.ch.Net.bind(Net.java:461)
E                   	at sun.nio.ch.Net.bind(Net.java:453)
E                   	at sun.nio.ch.ServerSocketChannelImpl.bind(ServerSocketChannelImpl.java:222)
E                   	at io.netty.channel.socket.nio.NioServerSocketChannel.doBind(NioServerSocketChannel.java:128)
E                   	at io.netty.channel.AbstractChannel$AbstractUnsafe.bind(AbstractChannel.java:558)
E                   	at io.netty.channel.DefaultChannelPipeline$HeadContext.bind(DefaultChannelPipeline.java:1283)
E                   	at io.netty.channel.AbstractChannelHandlerContext.invokeBind(AbstractChannelHandlerContext.java:501)
E                   	at io.netty.channel.AbstractChannelHandlerContext.bind(AbstractChannelHandlerContext.java:486)
E                   	at io.netty.channel.DefaultChannelPipeline.bind(DefaultChannelPipeline.java:989)
E                   	at io.netty.channel.AbstractChannel.bind(AbstractChannel.java:254)
E                   	at io.netty.bootstrap.AbstractBootstrap$2.run(AbstractBootstrap.java:364)
E                   	at io.netty.util.concurrent.AbstractEventExecutor.safeExecute(AbstractEventExecutor.java:163)
E                   	at io.netty.util.concurrent.SingleThreadEventExecutor.runAllTasks(SingleThreadEventExecutor.java:403)
E                   	at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:463)
E                   	at io.netty.util.concurrent.SingleThreadEventExecutor$5.run(SingleThreadEventExecutor.java:858)
E                   	at io.netty.util.concurrent.DefaultThreadFactory$DefaultRunnableDecorator.run(DefaultThreadFactory.java:138)
E                   	at java.lang.Thread.run(Thread.java:748)
```

This started failing in master after merging #2725, but was already failing before merging for the PR #2727. So seems like no change in our codebase caused this, but something that changed in our CI environment. 

Not sure if something changed in GitHub actions that can be causing the error. I don't know much about how Spark works to understand or debug the error.

I had a look at the dependencies to see if any version change may have caused this, but I don't see anything obvious:

| package                       | version working                | version failing              |
|:------------------------------|:--------------------------|:--------------------------|
| _libgcc_mutex                 | nan                       | 0.1                       |
| backports.functools_lru_cache | 1.6.3                     | 1.6.4                     |
| conda                         | 4.10.0                    | 4.10.1                    |
| conda-forge-pinning           | 2021.04.08.09.49.10       | 2021.04.15.09.30.17       |
| conda-package-handling        | 1.7.2                     | 1.7.3                     |
| curl                          | 7.76.0                    | 7.76.1                    |
| decorator                     | 5.0.6                     | 5.0.7                     |
| gdk-pixbuf                    | 2.42.4                    | 2.42.6                    |
| glib                          | 2.68.0                    | 2.68.1                    |
| glib-tools                    | 2.68.0                    | 2.68.1                    |
| ibis-framework                | 1.4.0+109.gd0202496.dirty | 1.4.0+110.ge0f2f0ab.dirty |
| importlib-metadata            | 3.10.0                    | 3.10.1                    |
| importlib_metadata            | 3.10.0                    | 3.10.1                    |
| libcurl                       | 7.76.0                    | 7.76.1                    |
| libglib                       | 2.68.0                    | 2.68.1                    |
| nbsphinx                      | 0.8.1                     | 0.8.3                     |
| pandas                        | 1.2.3                     | 1.2.4                     |
| prometheus_client             | 0.10.0                    | 0.10.1                    |
| sphinx                        | 3.5.3                     | 3.5.4                     |
| sqlalchemy                    | 1.4.6                     | 1.4.8                     |
| typed-ast                     | 1.4.2                     | 1.4.3                     |

CC: @jreback @icexelloss

--- a/ibis/backends/tests/base.py
+++ b/ibis/backends/tests/base.py
@@ -186,9 +186,11 @@ def get_common_spark_testing_client(data_directory, connect):
     import pyspark.sql.types as pt
     from pyspark.sql import SparkSession
 
-    spark = SparkSession.builder.config(
-        'spark.default.parallelism', 4
-    ).getOrCreate()
+    spark = (
+        SparkSession.builder.config('spark.default.parallelism', 4)
+        .config('spark.driver.bindAddress', '127.0.0.1')
+        .getOrCreate()
+    )
     _spark_testing_client = connect(spark)
     s = _spark_testing_client._session
     num_partitions = 4

